your-rents-services-geodata/src/main/java/com/yourrents/services/geodata/repository/ProvinceRepository.java

```diff
@@ -117,6 +121,16 @@ public boolean delete(UUID uuid) {
 				.where(PROVINCE.EXTERNAL_ID.eq(uuid))
 				.fetchOptional(PROVINCE.ID).orElseThrow(
 						() -> new DataNotFoundException("Province not found: " + uuid));
+		Result<Record1<Integer>> result = dsl.select(CITY.ID)
+				.from(CITY)
+				.where(CITY.PROVINCE_ID.eq(provinceId))
+				.limit(1)
+				.fetch();
+		if (result.isNotEmpty()) {
+			throw new IllegalArgumentException(
```

Comment:
It's not exactly a problem of wrong/illegal arguments.

What about adding a new type of exception (for example: DataOperationException), handled on the http side with a 409 status code?

Reply:
will change